lazy-pages/src/sys.rs

```diff
@@ -166,6 +173,14 @@ impl CheckedWasmAddr {
     }
 }
 
+pub struct DefaultUserSignalHandler;
+
+impl UserSignalHandler for DefaultUserSignalHandler {
```

Comment:
What the reason to use struct instead of function?

Reply:
And what the reason to add it as template param?

Reply:
> What the reason to use struct instead of function?

Because you cannot just pass function pointer in generic

> And what the reason to add it as template param?

Because I don't want to spend additional resources for something like TLS

Reply:
Can you please explain more widely cannot currently see any additional resources usages

Reply:
> currently

That's why I use generics. You ask me:

> And what the reason to add it as template param?

So I thought you mean it should be done in an another way. For example, thread local storage, which gives some overhead

Reply:
Currently we have only one `DefaultSignalHandler` ? What the reason to make it as generic param ? Even if we will have two signal handlers in future they can have different args or returns

Reply:
> What the reason to make it as generic param ?

Tests

Reply:
Ok see now